packages/react-bindings/src/contexts/StreamVideoContext.tsx

```diff
@@ -36,26 +36,9 @@ export const StreamVideo = ({
   language,
   translationsOverrides,
 }: PropsWithChildren<StreamVideoProps>) => {
-  const prevClient = useRef<StreamVideoClient | undefined>(undefined);
-
-  useEffect(() => {
-    if (client.user) {
-      client
-        .connectUser()
-        .catch((error) =>
-          console.error('Failed to establish connection', error),
-        );
-    }
-
-    prevClient.current = client;
-
-    return () => {
-      prevClient.current
-        ?.disconnectUser()
-        .catch((error) => console.error(`Failed to disconnect`, error));
-    };
-  }, [client]);
-
+  if (!client) {
+    return null;
+  }
```

Comment:
I'd rather lift this to the apps. This way, it is a bit implicit.

Reply:
I did this to avoid having to do this for all apps: 

```
{client && <StreamVideo>{call && <StreamCall></StreamCall>}</StreamVideo>}
```

It was lots of extra work since both `client` ans `call` are `undefined` initially

Reply:
I worry if we leave it like this, we would be making a huge breaking change in case we stop accepting "undefined" `client` or `call` in those providers.

Reply:
Yes, if we change this, that'd be a breaking change. But since we don't have plans for doing that at the moment, I think it's better to live with this convenience than to remove it just because we might break this in the future.